code/code/task/task_picklock.cc

```diff
@@ -1,144 +1,122 @@
+    ch->stopTask();
+    return;
   }
-  if ((exitp->lock_difficulty >= 100) || (difficulties > skill)) {
-    if ((difficulties > (skill + 100)) && !ch->isDextrous()) {
-      act("Uhoh, $n seems to have jammed the lock!",TRUE,ch,0,0,TO_ROOM);
-      ch->sendTo("Uhoh.  You seemed to have jammed the lock!\n\r");
-      exitp->lock_difficulty = min(100,exitp->lock_difficulty+10);
-      ch->stopTask();
-      return;
-    } else if (!::number(0,2)) {
-      act("You wiggle $p in the lock, without luck. (yet...)",
-          TRUE,ch,this,0,TO_CHAR);
-    }
-  } else if (ch->bSuccess(skill, SKILL_PICK_LOCK)) {
-    // this used to just be an automatic for the else case
-    // but what the heck, this makes it a bit harder to pick
-    // and helps make pick-lock a learn-by-doing too
-
-    REMOVE_BIT(exitp->condition, EXIT_LOCKED);
-    if (!exitp->keyword.empty()) {
-      act(format("$n skillfully picks the lock of the %s.\n\r") % 
-	  fname(exitp->keyword),
-	  TRUE,ch,0,0,TO_ROOM);
-    } else
-      act("$n picks the lock.", TRUE, ch, 0, 0, TO_ROOM);
+
+  if (pickable && ch->bSuccess(skill, SKILL_PICK_LOCK)) {
+    REMOVE_BIT(exit->condition, EXIT_LOCKED);
+    sstring msg = !exit->keyword.empty() ?
+        sstring(format("$n skillfully picks the lock of the %s.") % fname(exit->keyword)) :
+        sstring("$n picks the lock.");
+    act(msg, true, ch, nullptr, nullptr, TO_ROOM);
     ch->sendTo("The lock quickly yields to your skills.\n\r");
 
     // now for unlocking the other side, too
-    temp_rp = real_roomp(exitp->to_room);
-    if (temp_rp &&
-
-        (back = temp_rp->dir_option[rev_dir[ch->task->status]]) &&
+    TRoom *other = real_roomp(exit->to_room);
+    roomDirData *back;
+    if (other &&
+        (back=other->dir_option[rev_dir[ch->task->status]]) &&
         back->to_room == ch->in_room)
       REMOVE_BIT(back->condition, EXIT_LOCKED);
     ch->stopTask();
+    return;
   }
+
+  if (!::number(0,2))
+    act("You wiggle $p in the lock, without luck. (yet...)", true, ch, tool, nullptr, TO_CHAR);
 }
 
+
 int task_picklock(TBeing *ch, cmdTypeT cmd, const char *, int pulse, TRoom *, TObj *)
 {
-  TThing *pick;
-
   if (ch->isLinkdead() || (ch->getPosition() <= POSITION_SITTING)) {
     ch->stopTask();
     return FALSE;
   }
-  if (ch->utilityTaskCommand(cmd) ||
-      ch->nobrainerTaskCommand(cmd))
+
+  if (ch->utilityTaskCommand(cmd) || ch->nobrainerTaskCommand(cmd))
     return FALSE;
 
   if (ch->in_room != ch->task->wasInRoom) {
     ch->sendTo("Hey, where'd the lock go?!?\n\r");
-    act("$n looks about confused and bewildered.",TRUE,ch,0,0,TO_ROOM);
+    act("$n looks about, confused and bewildered.",TRUE,ch,0,0,TO_ROOM);
     ch->stopTask();
     return FALSE;
   }
+
   if (!ch->doesKnowSkill(SKILL_PICK_LOCK)) {
     ch->sendTo("I bet you wish you knew how to pick locks.\n\r");
     ch->stopTask();
     return FALSE;
   }
-  int pulses_to_wait;
+
   switch(cmd) {
     case CMD_TASK_CONTINUE:
+      TThing *pick;
```

Comment:
It'd look nicer to initialize while creating the variable, then check value

Reply:
I ended up doing it this way because of weird switch scoping rules I changed it as you describe, but had to wrap the whole case body inside of brackets to do it. otherwise C++ doesn't let you declare and initialize a variable on the same line inside a switch, because the variable's visibility falls through to further cases. why the original code was actually legal is kind of baffling though, since it's functionally equivalent to the illegal code. oh well, fixed!